internal/engine/pod_watch.go

```diff
@@ -126,19 +166,80 @@ func (w *PodWatcher) removeWatch(toRemove PodWatch) {
 	}
 }
 
-func (w *PodWatcher) triagePodUpdate(pod *v1.Pod) model.ManifestName {
-	manifestName := model.ManifestName(pod.ObjectMeta.Labels[k8s.ManifestNameLabel])
-	if manifestName != "" {
-		return manifestName
+// When new UIDs are deployed, go through all our known pods and dispatch
+// new events. This handles the case where we get the Pod change event
+// before the deploy id shows up in the manifest, which is way more common than
+// you would think.
+func (w *PodWatcher) setupNewUIDs(ctx context.Context, st store.RStore, newUIDs map[types.UID]model.ManifestName) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	for uid, mn := range newUIDs {
+		w.knownDeployedUIDs[uid] = mn
+
+		pod, ok := w.knownPods[uid]
+		if ok {
+			st.Dispatch(NewPodChangeAction(pod, mn))
+			continue
+		}
+
+		descendants := w.knownDescendentPodUIDs[uid]
+		for podUID := range descendants {
+			pod, ok := w.knownPods[podUID]
+			if ok {
+				st.Dispatch(NewPodChangeAction(pod, mn))
+			}
+		}
+	}
+}
+
+func (w *PodWatcher) triagePodUpdate(pod *v1.Pod, objTree k8s.ObjectRefTree) model.ManifestName {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	uid := pod.UID
+	oldPod, ok := w.knownPods[uid]
+
+	// In "real" code, if we get two pod updates with the same resource version,
+	// we can safely ignore the new one. But dispatching a spurious event
```

Comment:
do you mean "NOT dispatching a spurious event"?

Reply:
nope! this code dispatches an event when oldPod.ResourceVersion == pod.ResourceVersion

Reply:
OH! at a glance, the code looked like it was accounting for the case where a UID changed manifest associations o_0